### PR TITLE
Do not pull up projects that have set returning functions [#119042413]

### PR DESCRIPTION
**Pull request by d**

### CODE REVIEW

**Comment by hsyuan** [comment]
LGTM

File: libgpopt/src/operators/CNormalizer.cpp [hunk 1]
```diff
@@ -1246,7 +1246,10 @@ CNormalizer::PexprPullUpProjectElements
 		// a proj elem can be pulled up only if the defined column is not in
 		// pcrsUsed and its used columns are in pcrOutput
 		pexprPrEl->AddRef();
-		if (!pcrsUsed->FMember(pcrDefined) && pcrsOutput->FSubset(pcrsUsedByProjElem))
+
```
Comment (vraghavan78) [inline libgpopt/src/operators/CNormalizer.cpp]: can you update the comment about to state that we should not pull up set returning functions.

File: server/include/unittest/gpopt/minidump/CPullUpProjectElementTest.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#ifndef GPOPT_CPullUpProjectElementTest_H
```
Comment (vraghavan78) [inline server/include/unittest/gpopt/minidump/CPullUpProjectElementTest.h]: no header?

File: server/CMakeLists.txt [hunk 3]
```diff
@@ -87,7 +87,9 @@ add_executable(gporca_test
                src/unittest/gpopt/metadata/CTableDescriptorTest.cpp
                src/unittest/gpopt/metadata/CColumnDescriptorTest.cpp
                src/unittest/gpopt/metadata/CIndexDescriptorTest.cpp
-               src/unittest/gpopt/metadata/CNameTest.cpp)
+               src/unittest/gpopt/metadata/CNameTest.cpp
+               src/unittest/gpopt/minidump/CPullUpProjectElementTest.cpp
+        )
```
Comment (vraghavan78) [inline server/CMakeLists.txt]: Guys, we need to 
add_orca_test(CPullUpProjectElementTest)

To ensure that it runs

**Comment by vraghavan78** [comment]
+1

File: libgpopt/src/operators/CNormalizer.cpp (hunk 1 above)
Comment (hsyuan) [inline libgpopt/src/operators/CNormalizer.cpp]: sure, will do

File: server/include/unittest/gpopt/minidump/CPullUpProjectElementTest.h (hunk 2 above)
Comment (d) [inline server/include/unittest/gpopt/minidump/CPullUpProjectElementTest.h]: addressed in a new push

File: server/CMakeLists.txt (hunk 3 above)
Comment (d) [inline server/CMakeLists.txt]: duh, just did. that was dumb of me

File: libgpopt/src/operators/CNormalizer.cpp (hunk 1 above)
Comment (d) [inline libgpopt/src/operators/CNormalizer.cpp]: addressed in new commit 52030fd

**Pull request merged**